### PR TITLE
Refactor metadata handling

### PR DESCRIPTION
**Pull request by adambarreiro**
This PR contains #851, merge that one first.

## Description

The idea comes from [this suggestion](https://github.com/vmware/terraform-provider-vcd/pull/851#discussion_r872098181) made by @vbauzysvmware in #851.

The outcome is a cleaner code, with several duplications for `metadata` handling being removed and reduced to a single common place.

The PR requires https://github.com/vmware/go-vcloud-director/pull/473 as this one provides the new `MergeMetadata` function that is now used by the generic function.

## Extra
Bumps SDK to v2.17.0 to avoid [the mentioned bug](https://github.com/hashicorp/terraform-plugin-sdk/releases/tag/v2.16.0) when executing tests.

## Testing

Tested on VCD 10.3.2, 10.3.0, 10.2.2

### CODE REVIEW

**Review by vbauzysvmware — APPROVED** [review]
LGTM. Quite a save of cod lines :)

**Review by lvirbalas — CHANGES_REQUESTED** [review]

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,9 @@ go 1.13
 
 require (
 	github.com/hashicorp/go-version v1.4.0
-	github.com/hashicorp/terraform-plugin-sdk/v2 v2.14.0
+	github.com/hashicorp/terraform-plugin-sdk/v2 v2.16.0
```
[the remark this answers is not in the document]
Comment (adambarreiro) [reply go.mod:7]: Done. Also fixed a wrong entry in the changelog (853 was 843).
Not sure if the entry of this one is explanatory enough, let me know if you think it needs improvement!

**Comment by adambarreiro** [comment]
Bumped again SDK to 2.1.7.0

File: .changes/v3.7.0/853-improvements.md [hunk 2]
```diff
@@ -1,2 +1 @@
-* `resource/vcd_vm_sizing_policy`: remove (deprecate) unneeded `org` property [GH-853]
-* `datasource/vcd_vm_sizing_policy`: remove (deprecate) unneeded `org` property [GH-853]
+* `metadata` attribute on every compatible resource and data source is now more performant when adding and updating metadata [GH-853]
```
Comment (lvirbalas) [inline .changes/v3.7.0/853-improvements.md:1]: > is now more performant

You mean refactoring increased speed? How come? :)

**Review by mikeletux — APPROVED** [review]

File: .changes/v3.7.0/853-improvements.md (hunk 2 above)
Comment (adambarreiro) [reply .changes/v3.7.0/853-improvements.md:1]: Before we had a loop that iterated over all the metadata set. So imagine having a resource with a lot of metadata entries. Now, with the merge operation present in Go SDK is just one call, no looping required.

(Although for deletion we are still looping, as there is not an equivalent call for deleting all metadata at once)

Comment (adambarreiro) [reply .changes/v3.7.0/853-improvements.md:1]: It's true that "more performant" can be too optimistic as I didn't check it empirically 😅

**Review by lvirbalas — APPROVED** [review]

**Pull request merged**